docs/reference/images/kNN/knn_diagram.svg

```diff
@@ -0,0 +1 @@
0l2.3125 6.40625q0.4375 1.21875 0.796875 2.5625q0.328125 -1.296875 0.765625 -2.515625l2.375 -6.453125l1.875 0l-4.21875 11.25q-0.671875 1.828125 -1.046875 2.515625q-0.5 0.921875 -1.15625 1.34375q-0.640625 0.4375 -1.53125 0.4375q-0.546875 0 -1.203125 -0.21875z" fill-rule="nonzero"/><path fill="#ffffff" d="m387.8507 216.95287l-4.21875 -11.0625l1.984375 0l2.375 6.625q0.390625 1.078125 0.703125 2.234375q0.25 -0.875 0.703125 -2.109375l2.453125 -6.75l1.9375 0l-4.1875 11.0625l-1.75 0zm15.1640625 -3.5625l1.921875 0.234375q-0.453125 1.703125 -1.6875 2.640625q-1.234375 0.9375 -3.171875 0.9375q-2.421875 0 -3.84375 -1.484375q-1.421875 -1.5 -1.421875 -4.203125q0 -2.796875 1.4375 -4.328125q1.4375 -1.546875 3.734375 -1.546875q2.21875 0 3.609375 1.515625q1.40625 1.5 1.40625 4.25q0 0.15625 0 0.5l-8.25 0q0.109375 1.8125 1.03125 2.78125q0.921875 0.96875 2.3125 0.96875q1.03125 0 1.75 -0.53125q0.734375 -0.546875 1.171875 -1.734375zm-6.15625 -3.03125l6.171875 0q-0.125 -1.390625 -0.703125 -2.09375q-0.90625 -1.078125 -2.328125 -1.078125q-1.296875 0 -2.171875 0.859375q-0.875 0.859375 -0.96875 2.3125zm17.65857 2.546875l1.84375 0.234375q-0.296875 1.90625 -1.546875 2.984375q-1.25 1.078125 -3.0625 1.078125q-2.265625 0 -3.65625 -1.484375q-1.375 -1.484375 -1.375 -4.25q0 -1.796875 0.59375 -3.140625q0.59375 -1.34375 1.8125 -2.015625q1.21875 -0.671875 2.640625 -0.671875q1.796875 0 2.9375 0.921875q1.15625 0.90625 1.484375 2.578125l-1.828125 0.28125q-0.265625 -1.109375 -0.921875 -1.671875q-0.65625 -0.5625 -1.59375 -0.5625q-1.421875 0 -2.3125 1.015625q-0.875 1.015625 -0.875 3.203125q0 2.234375 0.84375 3.25q0.859375 1.0 2.234375 1.0q1.109375 0 1.84375 -0.671875q0.734375 -0.671875 0.9375 -2.078125zm7.5390625 2.375l0.265625 1.65625q-0.78125 0.15625 -1.40625 0.15625q-1.03125 0 -1.59375 -0.3125q-0.5625 -0.328125 -0.796875 -0.84375q-0.21875 -0.53125 -0.21875 -2.21875l0 -6.375l-1.375 0l0 -1.453125l1.375 0l0 -2.734375l1.859375 -1.125l0 3.859375l1.890625 0l0 1.453125l-1.890625 0l0 6.46875q0 0.8125 0.09375 1.046875q0.109375 0.21875 0.328125 0.359375q0.234375 0.125 0.640625 0.125q0.3125 0 0.828125 -0.0625zm1.1287537 -3.859375q0 -3.078125 1.71875 -4.546875q1.421875 -1.234375 3.46875 -1.234375q2.28125 0 3.71875 1.5q1.453125 1.484375 1.453125 4.125q0 2.140625 -0.640625 3.359375q-0.625 1.21875 -1.859375 1.90625q-1.21875 0.671875 -2.671875 0.671875q-2.3125 0 -3.75 -1.484375q-1.4375 -1.5 -1.4375 -4.296875zm1.9375 0q0 2.125 0.921875 3.1875q0.921875 1.046875 2.328125 1.046875q1.40625 0 2.328125 -1.0625q0.921875 -1.0625 0.921875 -3.234375q0 -2.046875 -0.9375 -3.109375q-0.921875 -1.0625 -2.3125 -1.0625q-1.40625 0 -2.328125 1.0625q-0.921875 1.046875 -0.921875 3.171875zm10.611694 5.53125l0 -11.0625l1.6875 0l0 1.671875q0.640625 -1.171875 1.1875 -1.546875q0.546875 -0.375 1.203125 -0.375q0.953125 0 1.921875 0.609375l-0.640625 1.734375q-0.6875 -0.40625 -1.375 -0.40625q-0.609375 0 -1.109375 0.375q-0.484375 0.359375 -0.6875 1.015625q-0.3125 1.0 -0.3125 2.1875l0 5.796875l-1.875 0z" fill-rule="nonzero"/></g></svg>
```

Comment:
cool picture!

Reply:
Thanks for the feedback. I went ahead and removed this illustration. While a visual is a "nice-to-have," I don't think it's crucial here.